tests/unit/commands/test_expand_extensions.py

```diff
@@ -63,6 +51,18 @@
         override-prime: |-
           craftctl default
           mkdir -p var/lib/pebble/default/layers
+    platforms:
+      amd64:
+        build_on: null
+        build_for: null
+    build-base: ubuntu:22.04
```

Comment:
this is a side-effect of the migration that we should consider: Because the project has two sets of fields (one from the base class and one from Rockcraft's class), the order of the exported fields changes and becomes a bit "unnatural". For example, `base` is now far away from `build-base` because the former comes from the superclass and the latter comes from the subclass.

Reply:
We could change the craft-application `Project` to not have any but the most basic values (name, title, summary, description) and then use an `__init_subclass__()` method to check for the existence of the other necessary attributes. This would require the application to define those attributes, but would ensure that the order is defined by the application as well.

Reply:
Hmmm maybe
@sergiusens what do you think? Is the "bad ordering" a big problem?

Reply:
I find the underscore build_on and build_for more disturbing. In general, this is not a problem we need to solve now

Reply:
Can we use `by_alias=True` to solve the underscore problem?